applications/machine_learning/src/modules/ml_runner.c

```diff
@@ -184,8 +204,10 @@ static bool handle_sensor_event(const struct sensor_event *event)
 static bool handle_ml_state_event(const struct ml_state_event *event)
 {
 	if ((event->state == ML_STATE_MODEL_RUNNING) && (state == STATE_SUSPENDED)) {
-		start_prediction();
-		state = STATE_ACTIVE;
+		state = current_active_state();
+		if (state == STATE_ACTIVE) {
+			start_prediction();
+		}
 	} else if ((event->state != ML_STATE_MODEL_RUNNING) && (state == STATE_ACTIVE)) {
```

Comment:
Handle `STATE_READY` too

Reply:
Should I really? I am starting new prediction only if the current state is ACTIVE. Ignoring it if the current state is READY. Should I call buf_cleanup if in ready state?

Reply:
I think so. In general we should always ensure that buffer is clean when we resume the prediction. Otherwise sensor data in the buffer will not be consistent (there will be a time gap). I prefer to avoid making any assumptions regarding relation between suspending the module and subscribing/unsubscribing (that could cause problems in the future).

Reply:
You are doing `buf_cleanup` every time you leave `STATE_ACTIVE`. That's good.
If you are in `STATE_READY` you are not setting module to suspended state. If a module enables subscription, the prediction will be started even through the module is suspended.

Reply:
No it would not - state would be set to the value returned by current_active_state - that may be ACTIVE or READY. The prediction is started only if the state after that is ACTIVE.

Reply:
Still, we will leave suspended state.

Reply:
Ok... Now I get it. So it is not the issue that we should reset it, but we should set STATE_SUSPENDED. And we are not leaving suspended state unintentionally but we are never going into it when expected. Good catch indeed.

Reply:
Fixed?